### PR TITLE
Duplicating Activity Player activities

### PR DESCRIPTION
**Pull request by emcelroy**
PT Story: https://www.pivotaltracker.com/story/show/176057523

[#176057523]

### CODE REVIEW

File: spec/support/shared_examples/remote_duplicate_support.rb [hunk 1]
```diff
@@ -41,7 +48,7 @@
       end
 
       def make_request
-        post :remote_duplicate, id: resource.id, user_email: user_email, add_to_portal: portal_url
+        post :remote_duplicate, id: resource.id, user_email: user_email, add_to_portal: portal_url, author_url: author_url
```
Comment (scytacki) [inline spec/support/shared_examples/remote_duplicate_support.rb:51]: This should only send the author_url when the home controller is being tested.
I'd suggest removing `let(:author_url)` from this file. 
And moving the `let(:aruthor_url)...` up into `it_behaves_like "remote duplicate support" do` block in the home_controller_spec.

I think if the author_url is undefined or nil then this line in make_request won't send the parameter. But you might need to change it so it author_url key isn't even in the hash send to the post method.

**Review by scytacki — CHANGES_REQUESTED** [review]
Overall looks good. I found an issue with the tests though where they aren't correctly exercising all of the code.

File: app/controllers/remote_duplicate_support.rb [hunk 2]
```diff
@@ -8,14 +8,29 @@ def self.included klass
...
+      resource_path = URI.parse(params[:author_url]).path
+      resource_type = resource_path.split("/")[1]
+      resource_id = resource_path.split("/")[2]
+      if resource_type == 'sequences'
+        resource = Sequence.find(resource_id)
+      else
+        resource = LightweightActivity.find(resource_id)
+      end
+    else
+      resource = @activity || @sequence
```
[the remark this answers is not in the document]
Comment (emcelroy) [reply app/controllers/remote_duplicate_support.rb:30]: Thanks. I believe this is fixed now.

**Review by scytacki — APPROVED** [review]
Looks good.

**Pull request merged**